reefperf/cloud_node.py

```
@@ -0,0 +1,48 @@
+from abc import abstractmethod, ABCMeta
+from reefperf.node_connection_provider import NodeConnectionProvider
+
+
+class CloudNode(object):
+    __metaclass__ = ABCMeta
+
+    @property
+    @abstractmethod
+    def connection(self):
+        pass
+
+# LC is a short for apache libcloud. It means that
+# all classes with prefix LC using this libcloud for operating on cloud nodes
+
+
+class LCCloudScaleNode(CloudNode):
+    def __init__(self, lc_node_obj, conn_cfg):
+        self._lc_node_obj = lc_node_obj
+        self._possible_usernames = [
+            'ubuntu', 'core', 'gentoo', 'centos',
+            'debian', 'arch', 'fedora'
+        ]
+        self._conn_cfg = conn_cfg
+        self._connection_obj = None
+
+    @property
+    def ipv4(self):
+        return self._lc_node_obj.public_ips[0]
+
+    @property
+    def username(self):
```

it returns `None` silently in case of a mismatch. Should raise an exception instead.